### PR TITLE
Minor performance improvements

### PR DESCRIPTION
**Pull request by roman-khimov**
Nothing radical here, nitpicking mostly. I guess we've lost some numbers during cache rework done earlier, but I'm not yet sure how to improve that.

### CODE REVIEW

File: pkg/core/native/std.go [hunk 1]
```diff
@@ -160,8 +160,8 @@ func newStd() *Std {
 	return s
 }
 
-func (s *Std) serialize(_ *interop.Context, args []stackitem.Item) stackitem.Item {
-	data, err := stackitem.Serialize(args[0])
+func (s *Std) serialize(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	data, err := ic.DAO.GetItemCtx().Serialize(args[0], false)
```
Comment (AnnaShaleva) [inline pkg/core/native/std.go:164]: After that we create ByteArray right from `data` without copying. Are you sure it's not dangerous? `stackitem.NewByteArray` returns a pointer to the provided array. We may reuse the same serialisation context during the subsequent instructions processing, so I'm afraid the resulting byte array may be corrupted.

File: pkg/util/uint160.go [hunk 2]
```diff
@@ -132,7 +132,12 @@ func (u *Uint160) UnmarshalJSON(data []byte) (err error) {
 
 // MarshalJSON implements the json marshaller interface.
 func (u Uint160) MarshalJSON() ([]byte, error) {
-	return []byte(`"0x` + u.StringLE() + `"`), nil
+	r := make([]byte, 3+Uint160Size*2+1)
```
Comment (AnnaShaleva) [inline pkg/util/uint160.go:135]: Such a meaningful time/alloc diff!

File: pkg/util/uint160.go [hunk 3]
```diff
@@ -132,7 +132,12 @@ func (u *Uint160) UnmarshalJSON(data []byte) (err error) {
...
+	copy(r, `"0x`)
+	r[len(r)-1] = '"'
+	slice.Reverse(u[:]) // u is a copy, so we can mangle it in any way.
+	hex.Encode(r[3:], u[:])
```
Comment (fyrchik) [inline pkg/util/uint160.go:139]: BTW, `String*` methods could be optimized in a similar way (~20% speedup on my machine).

File: pkg/core/native/std.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/core/native/std.go:164]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]
Looks good, but I still have doubdts about c3d989ebdaeada5107a96bf20aa2b4423b95c384, it's easy to forget about copying the result of serialisation if the context will be reused later. Have you benchmarked this commit, does it perform so good?

**Comment by roman-khimov** [comment]
It's good and most of the time serialization if followed by DB put where a copy is made, so it's not an issue except two cases where we copy explicitly.

**Pull request merged**